Mage.Sets/src/mage/cards/m/Musician.java

```diff
@@ -78,3 +65,66 @@ public Musician copy() {
     }
 }
 
+class MusicianEffect extends OneShotEffect {
+
+    MusicianEffect() {
+        super(Outcome.Benefit);
+        staticText = "If it doesn't have \"At the beginning of your upkeep, " +
+            "destroy this creature unless you pay {1} " +
+            "for each music counter on it,\" it gains that ability";
+    }
+
+    private MusicianEffect(final MusicianEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public MusicianEffect copy() {
+        return new MusicianEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Permanent permanent = game.getPermanent(source.getFirstTarget());
+        if (permanent == null) {
+            return false;
+        }
+        if(permanent.getAbilities().stream().anyMatch(ability -> ability instanceof MusicianTriggerAbility)){
+            return true;
+        }
+
+        OneShotEffect effect = new DoUnlessControllerPaysEffect(
+            new DestroySourceEffect(),
+            new DynamicValueGenericManaCost(
+                new CountersSourceCount(CounterType.MUSIC),
+                "{1} for each music counter on {this}"));
+        effect.setText("destroy this creature unless you pay {1} for each music counter on it");
+
+        game.addEffect(new GainAbilityTargetEffect(
+            new MusicianTriggerAbility(effect),
+            Duration.WhileOnBattlefield), source);
+        return true;
+    }
+}
+
+// This wrapper class is used to identify if the targetted creature already has the ability.
+class MusicianTriggerAbility extends BeginningOfUpkeepTriggeredAbility  {
+    public MusicianTriggerAbility(OneShotEffect effect) {
+        super(
+            Zone.BATTLEFIELD,
+            effect,
+            TargetController.YOU,
+            false,
+            false,
+            "At the beginning of your upkeep, ");
```

Comment:
shouldn't be necessary to manually specify the trigger string as the constructor without that parameter will generate it